### PR TITLE
Add ethers provider

### PR DESCRIPTION
**Pull request by ptescher**

### CODE REVIEW

File: bitski-provider/src/authenticated_ethers_provider.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            .network
+            .rpc_url
+            .parse()
+            .expect("Failed to parse RPC URL");
+
+        let transport = Http::new_with_client(url, client);
+        JsonRpcClient::request(&transport, method, params).await
+    }
+
+    // async fn convert_to_batch(
```
Comment (jtescher) [inline bitski-provider/src/authenticated_ethers_provider.rs:95]: probably don't need the comment

**Review by jtescher — APPROVED** [review]

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.68"
+channel = "1.72"
```
Comment (andschneider) [inline rust-toolchain.toml:2]: 1.73 is latest now

Comment (ptescher) [reply rust-toolchain.toml]: 👍

File: bitski/src/lib.rs [hunk 3]
```diff
@@ -114,16 +118,47 @@ impl Bitski {
...
+        let provider = self.get_web3_provider(network)?;
         Ok(Web3::new(provider))
     }
+
+    #[cfg(feature = "ethers")]
+    pub fn get_ethers_provider<N: TryInto<Network>>(
+        &self,
+        network: N,
+    ) -> Result<BitskiEthersProvider, Error> {
+    }
```
[the remark this answers is not in the document]
Comment (ptescher) [reply bitski/src/lib.rs]: fixed

File: bitski-provider/src/authenticated_ethers_provider.rs (hunk 1 above)
Comment (ptescher) [reply bitski-provider/src/authenticated_ethers_provider.rs]: 👍

**Pull request merged**